### DISCUSSION TITLE
Image files remain in file system after Image Attachment deletion

### DISCUSSION THREAD

**Discussion by maddin8** [netbox-community/netbox#16103] [Q&A] [answered]
I'm running Netbox 3.7.4 and didn't find notes about this in https://docs.netbox.dev/ or any Release Notes of Releases after 3.7.4:

When attaching an Image to a device, all works well and the file is created in /media/image-attachments/.
When deleting that Image Attachment from within the Netbox UI, the related file stays in /media/image-attachments/ and is seemingly not cleaned up by any housekeeping (I haven't found out yet how to check where housekeeping stores logs or other evidence that it ran when configured according to official documentation with cron).

The user as which Netbox runs in my environment (netbox) owns the "/media/image-attachments/" directory. Permissions for it are set to 755.

Did I do something wrong or is it intended that image files stay in the file system when the related Image Attachments are removed in the Netbox UI?

Thanks all for any hints!

**Comment by candlerb** [comment]
It works fine for me in Netbox 3.7.8

1. Attach image to device

2. Check filesystem

    ```
    root@netbox3:~# ls -l /opt/netbox/netbox/media/image-attachments/
    total 353
    -rw-r--r-- 1 netbox netbox 350569 May 13 12:18 device_99_c6qo7ehy91bl68x1683718473.jpg
    ```

3. Delete image from GUI:

    ![image](https://github.com/netbox-community/netbox/assets/44789/658fc015-7946-4a02-879f-fa84c38dd142)

4. Check filesystem again - it's gone immediately (no cron / background task required)

    ```
    root@netbox3:~# ls -l /opt/netbox/netbox/media/image-attachments/
    total 0
    ```

Note that I'm using the default storage location, whereas you say you're using `/media/image-attachments/`.

Try checking `journalctl -eu netbox` to see if any errors logged when attempting to delete the file.

**Comment by jeremystretch** [comment, marked as answer]
FWIW I can confirm that image attachment files should be deleted from disk automatically when the object is [deleted from the database](https://github.com/netbox-community/netbox/blob/4a64a3f6e0d0edf27996422eb2dbe0e197a6bea5/netbox/extras/models/models.py#L663).

**Comment by maddin8** [comment]
Thanks both! I'm glad must be an issue with my setup then, I will check it again.

@candlerb Thanks for mentioning the "journalctl" command, I will repeat the steps and check it out. The path to the image-attachments in my environment is "/opt/netbox/netbox/media/image-attachments/" as well, apologies for that I was unclear about this!

**Comment by maddin8** [comment]
I tried again and it worked multiple times without any further changes whatsoever. Unsure what caused this phenomenon the last time, maybe something else I don't know about was still accessing the file.

Anyway, thanks both for your help! 
If I encounter this again and can get error messages out of it, I'll post again.
